roles/matrix-server/templates/usr-local-bin/matrix-remove-all.j2

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+if [ "$(id -u)" != "0" ]; then
+        echo "This script must be executed as root! Aborting."
+        exit 1
+fi
+
+echo "WARNING! You are about to remove everything the playbook installs for matrix.{{ host_specific_hostname_identity }}: matrix, docker images,..."
+echo -n "If you're sure you want to do this, type: 'Yes, I really want to remove everything!'"
+read sure
+
+if [ "$sure" != "Yes, I really want to remove everything!" ]; then
+        echo "Good thing I asked, exiting"
+        exit 0
+else
+        echo "Stop and remove matrix services"
+        for s in $(find /etc/systemd/system/ -name "matrix-*" -printf "%f\n"); do
+                systemctl stop $s
+                rm -f /etc/systemd/system/$s
+        done
+        systemctl daemon-reload
+        echo "Remove matrix cronjobs"
+        find /etc/cron.d/ -name "matrix-*" -delete
+        echo "Remove matrix scripts"
+        find /usr/local/bin/ -name "matrix-*" -delete
+        echo "Remove every docker images"
+        docker rmi $(docker images -aq)
```

Comment:
Sure, that's why I put a warning at the beginning.
If you think it's too unsafe, maybe we should add another warning or an option.

Reply:
Indeed, it's a little bad how it deletes absolutely all images, but I don't see how we can do better (short of defining a long list of images to grep for; or skipping Docker image deletes completely).

Hmm.. I'm wondering how big of a problem this is going to be.. Usually, in-use images would be skipped and most images can be pulled again (it even happens automatically when starting them again), so it probably won't affect too many people in a way that they lose "data" (Docker images that can't easily be refetched/rebuilt again).

Reply:
I don't have much time these days to make a patch, but maybe a '--force' flag would be ok:
- if flag is put, we remove docker images
- if not, we leave docker images.

Reply:
Maybe if we are to add a flag, it should be named `--remove-docker-images` instead?

Or, since this script is interactive in any case, we can just omit the extra flag and ask for one more thing:

> This tool can delete the locally-stored Docker images too.
> It can't tell which Docker image was used by the playbook and which one is unrelated, so it goes for deleting ALL found images by default.
> Do you want all Docker images to be deleted (including ones that might not be related to this playbook) [yes/no]?